### PR TITLE
Filter invalid polygon shapes

### PR DESCRIPTION
**Pull request by eugene123tw**
### Summary

Filter invalid polygon shapes by checking points are within the image width and height.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/core/data/adapter/base_dataset_adapter.py [hunk 1]
```diff
@@ -272,11 +272,16 @@ def _prepare_label_information(
...
         """To filter out the abnormal polygon."""
         x_points = [annotation.points[i] for i in range(0, len(annotation.points), 2)]
         y_points = [annotation.points[i + 1] for i in range(0, len(annotation.points), 2)]
-        return min(x_points) < max(x_points) and min(y_points) < max(y_points)
+        return (
+            min(x_points) < max(x_points)
+            and min(y_points) < max(y_points)
+            and max(x_points) < width
+            and max(y_points) < height
+        )
```
Comment (chuneuny-emily) [inline src/otx/core/data/adapter/base_dataset_adapter.py:284]: Can the changed code conditions solve the OTXDetDataset: Invalid rectangle with coordinates: x1=1.0, y1=0.14171337069655723, x2=1.0, y2=0.19495596477181745 issue? Isn't this condition still False in min(x_points) < max(x_points)?

Comment (eugene123tw) [reply src/otx/core/data/adapter/base_dataset_adapter.py:284]: @chuneuny-emily The issue of invalidation occurs within the normalization in `_get_polygon_entity`. This occurs when the x or y coordinates exceed the dimensions of the image (e.g., image dimensions of 3500 x 3500, but points with coordinates such as (7000, 7000)). Consequently, this leads to the presence of invalid points in the following code:

```python
points=[
    Point(x=annotation.points[i] / width, y=annotation.points[i + 1] / height)
    for i in range(0, len(annotation.points), 2)
]
```

Comment (chuneuny-emily) [reply src/otx/core/data/adapter/base_dataset_adapter.py:284]: Okay. Thanks for explaining clearly. Then how about updating the code as below for the code readability?
`    x_min, x_max = min(x_points), max(x_points)
    y_min, y_max = min(y_points), max(y_points)

    has_area = x_min < x_max and y_min < y_max
    within_bounds = x_max < width and y_max < height

    return has_area and within_bounds`

**Review by goodsong81 — DISMISSED** [review]
LGTM for upper-bounding polygon points within the image size, but I'm not sure it also solves 0 size bounding box issue the Emily raised.

**Review by chuneuny-emily — DISMISSED** [review]

File: src/otx/core/data/adapter/base_dataset_adapter.py (hunk 1 above)
Comment (eugene123tw) [reply src/otx/core/data/adapter/base_dataset_adapter.py:284]: @chuneuny-emily I updated the condition for readability and added a area check.

**Review by goodsong81 — APPROVED** [review]

**Review by chuneuny-emily — APPROVED** [review]

**Review by yunchu — APPROVED** [review]

**Pull request merged**